File: solutions/javascript/secret-handshake/3/secret-handshake.js
export const commands = number => {
  const possibleActions = [
    'wink',
    'double blink',
    'close your eyes',
    'jump'
  ]
  
  const actions = possibleActions.filter((_, idx) => number & 2 ** idx)
  
  return number & 16 ? actions.reverse() : actions;
};

File: solutions/r/secret-handshake/1/secret-handshake.R
handshake <- function(n) {
  filter <- intToBits(n)[1:5] |>
    as.logical()
  
  actions <- c("wink", "double blink", "close your eyes", "jump")
  
  if (sum(filter[1:4]) == 0) NULL
  else if (filter[5]) rev(actions[filter[1:4]]) 
  else actions[filter[1:4]]
}